wfe/wfe.go

```diff
@@ -280,25 +360,41 @@ func (wfe *WebFrontEndImpl) handleKey(
 func (wfe *WebFrontEndImpl) Handler() http.Handler {
 	m := http.NewServeMux()
 	// GET only handlers
-	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, "GET")
+	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, false, "GET")
 	// Note for noncePath: "GET" also implies "HEAD"
-	wfe.HandleFunc(m, noncePath, wfe.Nonce, "GET")
-	wfe.HandleFunc(m, RootCertPath, wfe.handleCert(wfe.ca.GetRootCert()), "GET")
-	wfe.HandleFunc(m, rootKeyPath, wfe.handleKey(wfe.ca.GetRootKey()), "GET")
-	wfe.HandleFunc(m, intermediateCertPath, wfe.handleCert(wfe.ca.GetIntermediateCert()), "GET")
-	wfe.HandleFunc(m, intermediateKeyPath, wfe.handleKey(wfe.ca.GetIntermediateKey()), "GET")
+	wfe.HandleFunc(m, noncePath, wfe.Nonce, false, "GET")
+	wfe.HandleFunc(m, RootCertPath, wfe.handleCert(
+		func(no int) *core.Certificate {
+			return wfe.ca.GetRootCert(no)
+		},
+		wfe.ca.GetNumberOfRootCerts(), RootCertPath), true, "GET")
```

Comment:
Plumbing through the value of `GetNumberOfRootCerts` all the way to `handleCert` seems to add a lot of complexity. As far as I can tell it's only used to return 404 when a too-high number is requested. It would be simpler to rely on the fact that `GetRootCert` returns nil if the parameter is too high or too low. Then you don't need to define `GetNumberOfRootCerts` nor pass it as a parameter to `handleCert`.

Reply:
The reason I did this is because there are two distinct errors: 404 when a too-high (or invalid) number is requested, and 503 when the number is valid, but the certificate returned by the CA is `nil`. I assumed that this was because eventually, initialization of CA and WFE could be done in parallel, and then this would mean that the CA isn't fully initialized yet. If this isn't planned, returning always 404 and simplifying the code by not passing the value of `GetNumberOfRootCerts` around would be definitely easier.

Reply:
I can confirm we don't plan to initialize CA and WFE in parallel. Always returning 404 makes a lot of sense, thanks!

Reply:
Just noticed, the value of `GetNumberOfRootCerts` is still needed for calling `addAlternateLinks`, since it needs to know how many alternatives there are.